core/chains/evm/logpoller/orm.go

```diff
@@ -708,35 +712,33 @@ func (o *DbORM) blocksRangeAfterTimestamp(after time.Time, confs Confirmations,
 	if len(blocks) != 2 {
 		return 0, 0, nil
 	}
+	// If finalized logs are requested, return the minimum block_number and the maximum last_finalized_block_number.
+	// We can't rely on min(last_finalized_block_number), because we don't store this value from the beginning of the db table.
+	// We assume that in most of the cases min(block_number) <= max(last_finalized_block_number).
+	// min(block_number) > max(last_finalized_block_number) is possible, but it's a rare case.
+	// It happens only if timestamp is very close to now() and finality tag was just enabled for the chain.
+	//
+	// Example
+	// block_number | last_finalized_block_number | timestamp
+	// 100          | 0							  | 2021-01-01 00:00:00
+	// 101	        | 0							  | 2021-02-01 00:00:00
+	// 102	        | 90						  | 2021-03-01 00:00:00
+	// 103	        | 95						  | 2021-04-01 00:00:00
+	// In this case you will get 100 as the first block and 95 as the last finalized block.
```

Comment:
The only way to make it always reliable is to filter by `block_timestamp` in the outer query instead of here, but not sure if it's worth the effort